Adding `animate` method to Trajectory

prince-mathews opened the pull request:
- Within the ´animate´ method inside the Trajectory class, it wasn't possible to include ´stride´, ´atom_indices´, ´snapshot_indices´ or ´repeat´ arguments.

prince-mathews committed:
Update atomistic.py

- Within the ´animate´ method inside the Trajectory class, it wasn't possible to include ´stride´, ´atom_indices´, ´snapshot_indices´ or ´repeat´ arguments.

prince-mathews committed:
Update atomistic.py

pmrv reviewed (approved):
Looks good to me, thanks!

pmrv commented:
You had also code for the creation of a `Trajectory` from a list of structures, right?  Can you add that as well as a method to `Trajectory`?  Something like

```python
class Trajectory:
  ...
  @classmethod
  def from_structure_list(cls, structures):
     # somehow get all positions, etc. from all structures
     return cls(positions, ...)
```

prince-mathews commented:
> You had also code for the creation of a `Trajectory` from a list of structures, right? Can you add that as well as a method to `Trajectory`? Something like
> 
> ```python
> class Trajectory:
>   ...
>   @classmethod
>   def from_structure_list(cls, structures):
>      # somehow get all positions, etc. from all structures
>      return cls(positions, ...)
> ```

Isn't this something the Trajectory class does by itself?
I don't have a code, I just provide the Trajectory class the initial structure, positions of the structures I want to see in the trajectory and the Trajectory is created by the class itself.

pyiron-runner committed:
Format black

pmrv commented:
Mh, I was thinking really turning a list of `Atoms` objects into a trajectory.  But I remember now that the `Trajectory` class only works with "canonical" structures (equal number of particles everywhere).  I'll add some stuff to this branch.

pmrv committed:
Move animate to HasStructure

This also deprecates AtomisticGenericJob.animate_structure, because it
inherits animate_structure*s* from HasStructure now.  The older method
has a few more arguments, but it is implemented purely in terms of
calling

traj = job.trajectory(...)
traj.animate_structures(...)

Therefore users can replace calls of form

job.animate_structure(**traj_args, **anim_args)

very easily by

job.trajectory(**traj_args).animate_structures(**anim_args)

pmrv commented:
I've added some code to move `animate` over to `HasStructure`.  That means that a lot more jobs can now be animated (StructureContainer/TrainingContainer/StructureStorage/TrainingStorage/etc.).

pmrv committed:
Fix typo

pmrv committed:
Check for structure.cell only in job.animate_structure

self.structure is only available on jobs, so we do the check only in the
old animate method

pyiron-runner committed:
Format black
